integration_test/third_party_apps_test.go

```diff
@@ -237,6 +314,137 @@ func runLoggingTestCases(ctx context.Context, logger *logging.DirectoryLogger, v
 	return err
 }
 
+func runMetricsTestCases(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.VM, testCaseBytes []byte) error {
+	var metrics []expectedMetric
+	var err error
+	if err = yaml.UnmarshalStrict(testCaseBytes, &metrics); err != nil {
+		return fmt.Errorf("could not unmarshal contents of expected_metrics.yaml: %v", err)
+	}
+	if err = validateMetrics(metrics); err != nil {
+		return fmt.Errorf("expected_metrics.yaml failed validation: %v", err)
+	}
+	logger.ToMainLog().Printf("Parsed expected_metrics.yaml: %+v", metrics)
+	// Wait for the representative metric first, which is intended to *always*
+	// be sent. If it doesn't exist, we fail fast and skip running the other metrics;
+	// if it does exist, we go on to the other metrics in parallel, by which point they
+	// have gotten a head start and should end up needing fewer API calls before being found.
+	// In both cases we make significantly fewer API calls which helps us stay under quota.
+	for _, metric := range metrics {
+		if !metric.Representative {
+			continue
+		}
+		err = assertMetric(ctx, logger, vm, metric)
```

Comment:
Yeah, this was the result of a choice I made to have the test continue on to the non-representative metrics in case the representative one fails for some one-off non-missing reason. Alternatively we could just fail fast if the representative one fails for any reason, regardless of the error. Happy to add some comments if we stay with the current strategy though. WDYT?

Reply:
Current strategy sounds good. Just add a comment. See my suggested comment above.